azurerm/resource_arm_sql2017_elasticpool_test.go

```diff
@@ -0,0 +1,373 @@
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSqlElasticPool2017Exists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "per_database_settings.0.min_capacity", "0"),
+					resource.TestCheckResourceAttr(resourceName, "per_database_settings.0.max_capacity", "8"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMSqlElasticPool2017Exists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		serverName := rs.Primary.Attributes["server_name"]
+		poolName := rs.Primary.Attributes["name"]
+
+		client := testAccProvider.Meta().(*ArmClient).sql2017ElasticPoolsClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resp, err := client.Get(ctx, resourceGroup, serverName, poolName)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on sql2017ElasticPoolsClient: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: SQL2017 Elastic Pool %q on server: %q (resource group: %q) does not exist", name, serverName, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMSqlElasticPool2017Destroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).sql2017ElasticPoolsClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_sql2017_elasticpool" {
+			continue
+		}
+
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		serverName := rs.Primary.Attributes["server_name"]
+		poolName := rs.Primary.Attributes["name"]
+
+		resp, err := client.Get(ctx, resourceGroup, serverName, poolName)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("SQL2017 Elastic Pool still exists:\n%#v", resp.ElasticPoolProperties)
+		}
+	}
+
+	return nil
+}
+
+func testCheckAzureRMSqlElasticPool2017Disappears(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		serverName := rs.Primary.Attributes["server_name"]
+		poolName := rs.Primary.Attributes["name"]
+
+		client := testAccProvider.Meta().(*ArmClient).sql2017ElasticPoolsClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		_, err := client.Delete(ctx, resourceGroup, serverName, poolName)
+		if err != nil {
+			return fmt.Errorf("Bad: Delete on sql2017ElasticPoolsClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMSqlElasticPool2017_basic_DTU(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
```

Comment:
No blocker but we could pull out the perquisites into a template function and use that for all the different configs.